apps/evm/test/evm/operation/system_test.exs

```diff
@@ -21,4 +21,32 @@ defmodule EVM.Operation.SystemTest do
       assert Map.get(accounts, selfdestruct_address) == nil
     end
   end
+
+  describe "revert" do
+    test "halts execution reverting state changes but returning data and remaining gas" do
+      machine_code = MachineCode.compile([:push1, 3, :push1, 5, :revert, :push1, 10, :pop])
+      exec_env = %ExecEnv{machine_code: machine_code}
+      machine_state = %MachineState{program_counter: 0, gas: 24, stack: []}
+      substate = %SubState{}
+
+      {updated_machine_state, _, updated_exec_env, _} = VM.exec(machine_state, substate, exec_env)
```

Comment:
maybe it makes sense the check the output data too

Reply:
@vyorkin  done